ner_v1/detectors/numeral/number/number_detection.py

```diff
@@ -166,4 +200,87 @@ def _update_processed_text(self, original_number_strings):
             self.processed_text = self.processed_text.replace(detected_text, '')
 
+    def min_max_digit(self, min_digit, max_digit):
+        """
+        Update min max digit
+
+        Args:
+            min_digit (int): min digit
+            max_digit (int): max digit
+        """
+        self.min_digit = min_digit
+        self.max_digit = max_digit
+
+    def _detect_numerals(self):
+        """
+        1. Detects numerals from the given text
+        2. Makes a call to convert_numerals_to_numbers which return a tuple consisting of two lists
+            a.number_list consisting of numbers obtained from detected numerals
+            b.original_list consisting of original detected numerals
+
+        Returns:
+                (number_list, original_list) (tuple)
+                number_list (list): a list consisting of numbers obtained from detected numerals
+                original_list (list): a list consisting of detected numerals
+
+        Example:
+                self.text = "My favorite number is seven but jersey number is twenty five"
+                >>print(detect_numerals())
+                (['7', '25'], ['seven', 'twenty five'])
+        """
+        original_list = []
+        temp_str = ''
+        tokenizer = Tokenizer()
+
+        for word in tokenizer.tokenize(self.text.lower()):
+            if word in NUMERIC_VARIANTS:
+                temp_str = (temp_str + ' ' + word).strip()
+            else:
+                if temp_str.startswith('and') or (temp_str.endswith('and') and not temp_str.endswith('thousand')):
+                    temp_str = temp_str.replace('and', '')
+                original_list.append(temp_str.strip())
+                temp_str = ''
+        if temp_str.startswith('and') or (temp_str.endswith('and') and not temp_str.endswith('thousand')):
+            temp_str = temp_str.replace('and', '')
+        original_list.append(temp_str.strip())
+        original_list = [w for w in original_list if w != '']
+
+        for i in range(len(original_list)):
+            original_list_split = original_list[i].split()
+            original_list_split_len = len(original_list_split)
+            unit_count = len([unit for unit in original_list_split if unit in DIGIT_UNITS])
+
+            if original_list_split_len == unit_count:
+                original_list = original_list[:i] + original_list_split + original_list[i + 1:]
+
+        original_list_counter = 0
+        while original_list_counter < len(original_list):
+            number_len = len((str(w2n.word_to_num(str(original_list[original_list_counter])))))
+            if not self.max_digit >= number_len >= self.min_digit:
+                del original_list[original_list_counter]
+            original_list_counter = original_list_counter + 1
+        return self._convert_numerals_to_numbers(original_list)
+
+    @staticmethod
+    def _convert_numerals_to_numbers(original_list):
```

Comment:
Minor:
No need to make this static unless code outside this module is calling this